support/c/idris_system.c

```diff
@@ -2,5 +2,10 @@
 #include "idris_system.h"
 
 int idris2_system(const char* command) {
-    return system(command);
+    int status = system(command);
+    #ifdef _WIN32
+      return status;
+    #else
+      return WIFEXITED(status) ? WEXITSTATUS(status) : -1;
```

Comment:
I am unclear (perhaps you already know the answer) based on the documentation about whether `WIFEXITED()` will return 0 or non-zero when passed `-1`. At least one page I read indicated it might result in `255` when given `-1`.

The reason I ask, is because the documentation for `system()` indicates you get `-1` if the shell cannot be created or no exit status can be determined (and that `-1` is _without_ applying `WEXITSTATUS()`) so I wonder if we want to explicitly check for `-1` prior to doing the ternary you have here right now.

ref. https://man7.org/linux/man-pages/man3/system.3.html#RETURN_VALUE

Reply:
`WIFEXITED` returns `true` (i.e. non-zero) in the case when status is produced by calls to `exit` or `_exit`. Those functions cannot be used to make the status to be `-1` because only least significant 8 bits are passed according to documentation and `int` type cannot be less than two bytes according to the language spec. Also, if you test `WIFEXITED`  on `-1`, you expectedly get `0`. That said, I think the present check is sufficient.

Reply:
Sounds good. Thanks!